src/main/java/core/basesyntax/service/DataConverter.java

```
@@ -0,0 +1,8 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+
+public interface DataConverter {
+    List<FruitTransaction> convertToTransaction(List<String> rawLines);
```

The interface method signature is fine, but ensure implementations validate inputs. The checklist requires non-null parameter validation and informative RuntimeExceptions; DataConverterImpl currently returns an empty list for null input and lacks explicit checks for empty fruit name and negative quantities. Update DataConverterImpl to throw informative FruitShopException for null rawLines, empty fruit names, and negative quantities instead of silently returning or producing generic errors .